rs/nns/integration_tests/src/cycles_minting_canister_with_exchange_rate_canister.rs

```diff
@@ -489,3 +496,118 @@ fn test_disabling_and_reenabling_exchange_rate_canister_calling_via_exchange_rat
         cmc_first_rate_timestamp_seconds + FIVE_MINUTES_SECONDS * 4 + 22
     );
 }
+
+fn verify_cmc_certified_data<Data>(
+    state_machine: &StateMachine,
+    certificate: Vec<u8>,
+    hash_tree: Vec<u8>,
+    label: &[u8],
+    data: Data,
+) where
+    Data: CandidType,
+{
+    // Verify the certificate with the IC root key.
+    let root_key = state_machine.root_key_der();
+    let certificate = Certificate::from_cbor(certificate.as_slice()).unwrap();
+    let current_time_nanos = state_machine.get_time().as_nanos_since_unix_epoch() as u128;
+    certificate
+        .verify(
+            CYCLES_MINTING_CANISTER_ID.as_ref(),
+            root_key.as_slice(),
+            &current_time_nanos,
+            &60_000_000_000, // 1 minute in nanoseconds
+        )
+        .unwrap();
+
+    // Verify the hash tree with the certificate.
+    let certified_data = certificate.tree.lookup_path([
+        b"canister",
+        CYCLES_MINTING_CANISTER_ID.get().as_slice(),
+        b"certified_data",
+    ]);
+    let certified_data = match certified_data {
+        LookupResult::Found(certified_data) => certified_data,
+        _ => panic!("Failed to find certified_data in certificate"),
+    };
```

Comment:
```rust
    // Verify the hash tree with the certificate.
    let certified_data = certificate.tree.lookup_path([
        b"canister",
        CYCLES_MINTING_CANISTER_ID.get().as_slice(),
        b"certified_data",
    ]);
    let certified_data = match certified_data {
        LookupResult::Found(certified_data) => certified_data,
        _ => panic!("Failed to find certified_data in certificate"),
    };
```

This part is not necessary because it is already checked by the `certificate.verify()` function, the same is true for the `verify_canister_sig()` function from `ic-signature-verification`.

Reply:
Similar to another thread, the `ic-signature-verification` can't be used here just yet.